### PR TITLE
Fix: multiple-upload allow re-uploading the same file, can be re-opened after cancelling the dialog

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Fixes:
- Couldn't open the dialog twice without reloading the page.
- Couldn't re-upload twice the same file without reloading the page (upload A, delete A, upload A)

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/data_source/MultipleDocumentsUpload.tsx [hunk 1]
```diff
@@ -32,69 +33,162 @@ export const MultipleDocumentsUpload = ({
...
   const [isBulkFilesUploading, setIsBulkFilesUploading] = useState<null | {
     total: number;
     completed: number;
   }>(null);
 
+  const close = useCallback(
+    (save: boolean) => {
+      // Clear the values of the file input
+      if (fileInputRef.current) {
+        fileInputRef.current.value = "";
```
Comment (Fraggle) [inline front/components/data_source/MultipleDocumentsUpload.tsx:47]: fix the ability to re-upload the same file.

File: front/components/data_source/MultipleDocumentsUpload.tsx [hunk 2]
```diff
@@ -32,69 +33,162 @@ export const MultipleDocumentsUpload = ({
 }: MultipleDocumentsUploadProps) => {
   const fileInputRef = useRef<HTMLInputElement>(null);
   const [isLimitPopupOpen, setIsLimitPopupOpen] = useState(false);
-  const [clicked, setClicked] = useState(false);
-  useEffect(() => {
-    if (isOpen && !clicked) {
-      fileInputRef.current?.click();
-      setClicked(true);
```
Comment (Fraggle) [inline front/components/data_source/MultipleDocumentsUpload.tsx:39]: blocked the ability to re-open.

File: front/components/data_source/MultipleDocumentsUpload.tsx [hunk 3]
```diff
@@ -32,69 +33,162 @@ export const MultipleDocumentsUpload = ({
...
 
-    return new Ok(null);
-  };
+  useEffect(() => {
+    const ref = fileInputRef.current;
+    ref?.addEventListener("cancel", handleFileInputBlur);
+    return () => {
+      ref?.removeEventListener("cancel", handleFileInputBlur);
+    };
+  }, [handleFileInputBlur]);
```
Comment (Fraggle) [inline front/components/data_source/MultipleDocumentsUpload.tsx:183]: handle cancelling the dialog

File: front/components/data_source/MultipleDocumentsUpload.tsx [hunk 4]
```diff
@@ -32,69 +33,161 @@ export const MultipleDocumentsUpload = ({
...
+        text: null,
+        source_url: undefined,
+        tags: [],
+        light_document_output: true,
+        upsert_context: null,
+        async: false,
+      };
+
+      try {
+        const res = await fetch(
```
Comment (flvndvd) [inline front/components/data_source/MultipleDocumentsUpload.tsx:76]: Why not moving this to the "swr" folder?

**Review by flvndvd — APPROVED** [review]

File: front/components/data_source/MultipleDocumentsUpload.tsx (hunk 4 above)
Comment (Fraggle) [reply front/components/data_source/MultipleDocumentsUpload.tsx:76]: out of scope (and I have more urgent matters) but i'll do it next time i have the opportunity.

**Pull request merged**